### PR TITLE
Added a new instruction

### PR DESCRIPTION
**Pull request by YeffyCodeGit**
h

### CODE REVIEW

File: ANFuck/src/main.h [hunk 1]
```diff
@@ -12,8 +12,7 @@ namespace ANFuck {
 		std::string commands;
 		std::ifstream file(filename);
 
-		if (!file.is_open())
-			throw std::runtime_error("Failed to open '" + filename + "'.");
+		if (!file.is_open()) throw std::runtime_error("Failed to open '" + filename + "'.");
```
Comment (ANF-Studios) [inline ANFuck/src/main.h:15]: There is no need for that. You have to keep in mind when contributing to open source projects that every maintainer has their own programming style and you should keep to that and keep it consistent.

File: ANFuck/src/main.h [hunk 2]
```diff
@@ -96,13 +96,10 @@ namespace ANFuck {
...
-					if (instructionStack.empty())
-						throw std::runtime_error("Found a ']' that did not have a matching '['!");
+					if (instructionStack.empty()) throw std::runtime_error("Found a ']' that did not have a matching '['!");
 
-					if (*dataPtr != 0)
-						instructionPtr = instructionStack.top();
-					else
-						instructionStack.pop();
+					if (*dataPtr != 0) instructionPtr = instructionStack.top();
+					else instructionStack.pop();
```
Comment (ANF-Studios) [inline ANFuck/src/main.h:102]: Revert these back as well please.

File: ANFuck/src/main.h [hunk 3]
```diff
@@ -117,7 +114,8 @@ namespace ANFuck {
 				}
 				case '*':
 				{
-					// TODO: Think of what this part should do.
+					*dataPtr = defaultDataPtr;
+					std::cout << *dataPtr;
```
Comment (ANF-Studios) [inline ANFuck/src/main.h:118]: Why are we printing a value?

**Review by ANF-Studios — CHANGES_REQUESTED** [review]
You can work on those suggested changes and, also, please show me a preview of it being used.

File: ANFuck/src/main.h (hunk 3 above)
Comment (YeffyCodeGit) [reply ANFuck/src/main.h:118]: That was mostly for testing to make sure if it was working, ill remove it now

**Comment by YeffyCodeGit** [comment]
h

**Comment by ANF-Studios** [comment]
CI fails, compiler errors, code non-functional.

**Comment by ANF-Studios** [comment]
Doesn't work as expected.
image

File: ANFuck/src/main.cpp [hunk 4]
```diff
@@ -47,7 +47,7 @@ int main(int argc, char* argv[]) {
 }
 
 void GetAndRunCode(const std::string& pathToFile) {
-	try
+    try
```
Comment (ANF-Studios) [inline ANFuck/src/main.cpp:50]: Bad formatting.

File: ANFuck/src/main.h [hunk 5]
```diff
@@ -96,12 +99,12 @@ namespace ANFuck {
...
 				{
-					if (instructionStack.empty())
+					if (instructionStack.empty()) 
 						throw std::runtime_error("Found a ']' that did not have a matching '['!");
 
-					if (*dataPtr != 0)
+					if (*dataPtr != 0) 
 						instructionPtr = instructionStack.top();
-					else
+					else
```
Comment (ANF-Studios) [inline ANFuck/src/main.h:107]: You added empty spaces again.

File: ANFuck/src/main.h [hunk 6]
```diff
@@ -126,6 +130,7 @@ namespace ANFuck {
 			instructionPtr++;
 		}
 
+
```
Comment (ANF-Studios) [inline ANFuck/src/main.h:133]: Unnecessary empty line

File: ANFuck/src/main.h [hunk 7]
```diff
@@ -117,7 +120,8 @@ namespace ANFuck {
 				}
 				case '*':
 				{
-					// TODO: Think of what this part should do.
+					// Set the dataPtr to what it was at the beggining
+					*dataPtr = *defaultDataPtr;
```
Comment (ANF-Studios) [inline ANFuck/src/main.h:124]: This doesn't work. Are you sure I'm not doing anything wrong?

**Review by ANF-Studios — CHANGES_REQUESTED** [review]
I'm not really confident that this pull even accomplishes what it targets.

**Comment by ANF-Studios** [comment]
@YeffyCodeGit, please do not just open and close issues.

**Pull request closed without merging**